include/swift/Sema/Constraint.h

```diff
@@ -852,7 +852,11 @@ class Constraint final : public llvm::ilist_node<Constraint>,
   /// Clone the given constraint.
   Constraint *clone(ConstraintSystem &cs) const;
 
-  void print(llvm::raw_ostream &Out, SourceManager *sm) const;
+  /// Prints constraint placed on type and constraint properties
+  ///
+  /// \c repeatedLocator checks if current locator is repeated in
+  /// preceding constraint and prevents repetitive locator printing.
+  void print(llvm::raw_ostream &Out, SourceManager *sm, bool repeatedLocator = false) const;
```

Comment:
How about we rename new parameter into `skipLocator`? `print` just need to to know that locator shouldn't be printed, it doesn't need to know nuances about why, that's for caller to determine.

Reply:
Got it! Should I also get rid of documentation or update it to reflect `skipLocator`?

Reply:
Please update it to describe what skipping locator actually means :)